internal/streaming/streaming.go

```diff
@@ -0,0 +1,32 @@
+package streaming
+
+import (
+	"github.com/cosmos/cosmos-sdk/codec"
+	servertypes "github.com/cosmos/cosmos-sdk/server/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	abci "github.com/tendermint/tendermint/abci/types"
+)
+
+// TOML configuration parameter keys
+const (
+	// TomlKey is the top-level TOML key for streaming configuration
+	TomlKey = "streaming"
+
+	// EnabledParam is the flag that enables ABCI request response streaming services
+	EnabledParam = "enabled"
+)
```

Comment:
If this is expected to be a custom addition to one of the config files, it's not going to jive well with our `config` commands. Specifically, the `set`, `pack`, and `unpack` commands will stomp out any such custom entry (since they're not part of the template).

Annoyingly, for the `plugins` config, it looks like they went with the key `on`, where their other sections use `enabled`, and then `plugins.enabled` as a slice of strings with the names of the plugins to use.

In my opinion, `enabled` should be a boolean indicating whether or not that feature should be active. Then something like `services` or `names` would be a slice of strings with the names of the plugins to use.

Regardless, as far as I can tell, this `streaming` config section is new. ADR-048 mentions a `plugins.streaming` section, but no `plugins.streaming.enabled` or `streaming.enabled` entries and no `streaming` section.

If any of this is new, and specific to provenanced (i.e. not generic or better defined by Cosmos-SDK), we should consider creating our own config file to house these things. If that's the case, we'll want to do that as a separate issue though. But either way, if this is a new config thing, there needs to be documentation on it.

Reply:
I borrowed from the plugin system in ADR-038. However, it is different because in this implementation we're not able to capture any state change events. The reason for this lighter imlementation is because adr-038 has an `AppHash` error that's being caused either how state-change events are pushed to the listeners or how the plugin system is injected. In a discussion with @iramiller I came up with this version.

This implementation and its configuration is custom to Provenance. So we can change the config names. The `streaming` section is new and has no relation to `plugins.streaming` from ADR-038. Per @iramiller we want to disable ADR-038 implementation we have in our Cosmos SDK fork and wait for upstream to merge ADR-038 (This has seen some progress as of the last two days).

For the config. I actually think its better we remove `enabled` all together. I don't think we should actually have more than one service enabled at a time. 
We would just have `[streaming]` and `streaming.service = kafka | postgresql, trace' etc. But only one enabled.  So the `BeginBlocker` and `EndBlocker` only push out to one external service. Than clients can consumer from that external service. I'll change the design to reflect this. 

I would love some help on figuring out the config so it does work with `provenance config set`.

Reply:
To add to this a little more I also have a `Kafka` implementation on as separate [branch](https://github.com/provenance-io/provenance/tree/951-abci-beginblocker-and-endblocker-streaming-to-kafka). I wanted to separate the API support from service implementations.

Reply:
> I would love some help on figuring out the config so it does work with provenance config set.

The `config` command stuff all works off of reflection on the config structs. Writing the config files is then done by providing the appropriate config struct to the appropriate toml template. Since none of the templates have these fields, they'll disappear any time the config file is programmatically updated.

I've been thinking for quite some time that it'd be nice if we had our own `provenance.toml` config file (and associated object). But the only thing I could think to put in there so far is the `testnet` flag. If we want this `streaming` config to be in a file, that's probably the best way to go.

We'd want to do that as a separate issue/PR though, so for this, we just need documentation about what the config fields are, and what's expected. That way, we know what to make our template/struct look like.

Also, I mentioned this in the Cosmos PR, but the standard boolean on/off field name is `enable`. I like having such a field because it allows you to turn something off without losing how you've configured it. So there'd be a `streaming.enable` boolean and `streaming.service` string.

Reply:
It would be interesting to see if there could be a way an app like provenance using the sdk/tendermint could create its own consolidated and documented toml file to replace all the others and simplify...

Reply:
Summary:

1. I've created #962 for the new config file.
2. The on/off field should be `enable` (not `enabled`).
3. Make sure the documentation (e.g. in `app/streaming/README.md`) is accurate.

Until #962 is completed, we'll just have to live with the knowledge that the `config` commands a) won't find these new fields, and b) might unconfigure this stuff.

Reply:
👍 I'll get started on the `proveance.toml`.